src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi

```diff
@@ -67,3 +67,23 @@ class _EOF:
 
 
 EOF = _EOF()
+
+
+class BaseError(Exception):
+    """The base class for all exceptions generated by gRPC framework."""
+
+
+class UsageError(BaseError):
+    """Raised when the usage might lead to undefined behavior."""
```

Comment:
Can we be more specific? *Whose* usage? Usage of *what*?

Reply:
Good suggestions. More details added:
```python
class UsageError(BaseError):
    """Raised when the usage of API by applications is inappropriate.

    For example, trying to invoke RPC on a closed channel, mixing two styles
    of streaming API on the client side. This exception should not be
    suppressed.
    """
```